### PR TITLE
Clarify properly why mouse buttons are not fully synchronised

### PR DESCRIPTION
**Pull request by frenzibyte**
As per #1682's description.

### CODE REVIEW

File: osu.Framework/Input/PassThroughInputManager.cs [hunk 1]
```diff
@@ -154,7 +154,8 @@ public void Sync(bool useCachedParentInputManager = false)
         /// <param name="parentState">Parent's state. If this is null, it is regarded as an empty state.</param>
         protected virtual void SyncInputState(InputState parentState)
         {
-            // release all buttons that is not pressed on parent state
+            // Only release buttons not pressed in parent, synchronizing what parent has already pressed
+            // as well may potentially lead to false mouse click input propagated to children.
```
Comment (bdach) [inline osu.Framework/Input/PassThroughInputManager.cs:158]: I can't parse this sentence.

How about:

```suggestion
            // invariant: if mouse button is pressed in child, then it is pressed in parent (but not the converse)
            // therefore, mouse up events are always synced from parent
            // mouse down events are not synced to prevent fake delayed clicks
```

Comment (frenzibyte) [reply osu.Framework/Input/PassThroughInputManager.cs:158]: I don't get what it means by `if mouse button is pressed in child, then it is pressed in parent (but not the converse)`, perhaps you got it swapped?

Comment (bdach) [reply osu.Framework/Input/PassThroughInputManager.cs:158]: I meant it precisely as written. Logically speaking we have the following possibilities:

| parent | child | possible |
| :-: | :-: | :-: |
| pressed | pressed | :heavy_check_mark: yes |
| pressed | not pressed | :heavy_check_mark: yes |
| not pressed | pressed | ❌ no |
| not pressed | not pressed | :heavy_check_mark: yes |

Therefore if the button is seen as pressed in the child then it must be pressed in the parent, but when it is not pressed, it's no longer ensured that it isn't pressed in the parent.

Comment (frenzibyte) [reply osu.Framework/Input/PassThroughInputManager.cs:158]: oh so you're referring `child` as the current state and `parent` as the new state, that thing got me confused there, maybe reword to:
```suggestion
            // invariant: if mouse button is currently pressed, then it has been pressed in parent (but not the converse)
            // therefore, mouse up events are always synced from parent
            // mouse down events are not synced to prevent false clicks
```

Comment (bdach) [reply osu.Framework/Input/PassThroughInputManager.cs:158]: sure
